extensions/sql-database-projects/src/test/dialogs/publishOptionsDialog.test.ts

```diff
@@ -12,13 +12,16 @@ import * as TypeMoq from 'typemoq';
 import { PublishOptionsDialog } from '../../dialogs/publishOptionsDialog';
 import { PublishDatabaseDialog } from '../../dialogs/publishDatabaseDialog';
 import { Project } from '../../models/project';
+import sinon = require('sinon');
 
 describe('Publish Database Options Dialog', () => {
 	before(async function (): Promise<void> {
 		await baselines.loadBaselines();
 	});
 
 	it('Should open dialog successfully ', async function (): Promise<void> {
+		const proj = new Project('');
```

Comment:
Earlier you changed something that was using new Project to use Project.openProject - what was the difference there that needed that to be different?

Reply:
I merged this PR in on Friday so that I can use the canary builds over the weekends to verify the stableness of the tests. 

even in a successful run, I saw the following errors in the log
![image](https://user-images.githubusercontent.com/13777222/177390708-55cabb51-0717-40d0-8f58-a7740c71c082.png)

it turns out to be caused by the project XML document being undefined, and the initialization is done in project.openProject, @kisantia, I think we should probably do the initialization in the constructor to avoid creating uninitialized project objects, there are still other test cases using the constructor directly, they are not failing because getProjectTargetPlatform (or other methods requires project xml being loaded) is not called during the test.

another change I made is to this test case: https://github.com/microsoft/azuredatastudio/commit/416e607f321cb01600d72538deaf89bf4e090ba8#diff-323242a13bb4742dd45d6242b0bc5258424fa6be7c86475a22f86f950ec94ce7L374

the dialog is initialized once but used twice, this is not correct because after the first use the dialog object will be disposed, the fix is to recreate the dialog object.